app/javascript/src/LoginPage/loginForms/SignupForm.jsx

```diff
@@ -53,68 +74,75 @@ const InputFormGroup = (props) => {
 }
 
 class SignupForm extends Component<Props, State> {
-  constructor (props: Props) {
-    super(props)
-    this.state = {
-      username: this.props.user.username,
-      emailAddress: this.props.user.email,
-      firstname: this.props.user.firstname,
-      lastname: this.props.user.lastname,
-      password: '',
-      passwordConfirmation: '',
-      isFilledUsername: undefined,
-      isFilledEmailAddress: undefined,
-      isFilledFirstname: true,
-      isFilledLastname: true,
-      isFilledPassword: undefined,
-      isFilledPasswordConfirmation: undefined
+  state = {
+    username: this.props.user.username,
+    email: this.props.user.email,
+    firstName: this.props.user.firstname,
+    lastName: this.props.user.lastname,
+    password: '',
+    passwordConfirmation: '',
+    validation: {
+      username: undefined,
+      email: undefined,
+      firstName: true,
```

Comment:
I don't think this is correct, validation should be `undefined` just like the rest, unless the field is filled in by default?